newIDE/app/src/ObjectsList/index.js

```diff
@@ -168,6 +171,21 @@ export default class ObjectsList extends React.Component<Props, State> {
     );
     object.setTags(getStringFromTags(this.props.selectedObjectTags));
 
+    const additionalService = ObjectsAdditionalService.getServices(
+      object,
+      objectsContainer,
+      false
+    );
+    if (additionalService) {
+      additionalService.onObjectAdded(object, project, objectsContainer);
+      const infoBarDetails = additionalService.getInfoBarDetails(
+        'onObjectAdded'
+      );
+      if (infoBarDetails) {
+        this.props.onAdditionalServiceComplete(infoBarDetails);
+      }
+    }
```

Comment:
Much like in InstancesEditor, see if you can run this in the SceneEditor?

Reply:
@4ian I guess we can handle everything in the SceneEditor itself? As I see it, we already have a "stream of events" flowing across different editors to SceneEditor (for eg. InstancesAdder to InstancesEditor to SceneEditor), whereas I
believe I'm exposing SceneEditor to other editors (ie, ObjectsList, InstancesEditor, etc) and introducing a flow in the opposite direction. My initial thought process was to declutter the SceneEditor, but I guess it is going to be cleaner even after adding code in SceneEditor (and making it a bit bulkier).

Reply:
Yep I agree that SceneEditor is becoming a bit cluttered/complex, but that is its nature: it's the component that is orchestrating the other editors :) 
Note that React hooks are a partial solution to this because they allow to break a component like this which is big with a lot of logic into smaller reusable logic chunks... unfortuantely for now it's still a class component!

But again it's fine to do this in the SceneEditor, being the receiver of the "stream of event" is basically its role. If something is too complex, it can always be extracted to a function and used by SceneEditor.